vault/raft.go

```diff
@@ -1230,9 +1232,18 @@ func (c *Core) raftLeaderInfo(leaderInfo *raft.LeaderJoinInfo, disco *discover.D
 			return nil, fmt.Errorf("failed to parse addresses from auto-join metadata: %w", err)
 		}
 		for _, ip := range clusterIPs {
-			if strings.Count(ip, ":") >= 2 && !strings.HasPrefix(ip, "[") {
+			if count := strings.Count(ip, ":"); count >= 2 && !strings.HasPrefix(ip, "[") {
 				// An IPv6 address in implicit form, however we need it in explicit form to use in a URL.
 				ip = fmt.Sprintf("[%s]", ip)
+			} else if count > 0 && !strings.HasSuffix(ip, "]") {
+				tmpIp, portStr, err := net.SplitHostPort(ip)
+				if err == nil {
+					ip = tmpIp
+					tmpPort, err := strconv.Atoi(portStr)
+					if err == nil {
+						port = uint(tmpPort)
+					}
+				}
```

Comment:
@aubrich would you be able to add unit testing for this? it might be interesting to isolate this in a function and unit test the function. this would give us more confidence we're not introducing bugs to the codebase.